src/extension/control-flow/src/main/java/org/geoserver/flow/config/DefaultControlFlowConfigurator.java

```diff
@@ -195,4 +202,53 @@ public long getTimeout() {
         return timeout;
     }
 
+    @Override
+    public List<Resource> getFileLocations() throws IOException {
+        List<Resource> configurationFiles = new ArrayList<>();
+        GeoServerResourceLoader loader = GeoServerExtensions.bean(GeoServerResourceLoader.class);
+        if (loader != null) {
+            Resource controlflow = loader.get(PROPERTYFILENAME);
+        
+            configurationFiles.add(controlflow);
+        } else if (this.configFile != null && this.configFile.getResource() != null) {
+            configurationFiles.add(this.configFile.getResource());
+        }
+        return configurationFiles;
+    }
+
+    @Override
+    public void saveConfiguration(GeoServerResourceLoader resourceLoader) throws IOException {
+        GeoServerResourceLoader loader = GeoServerExtensions.bean(GeoServerResourceLoader.class);
+        if (loader != null) {
+            for(Resource controlflow : getFileLocations()) {
+                Resource targetDir = 
+                        Files.asResource(resourceLoader.findOrCreateDirectory(Paths.convert(loader.getBaseDirectory(), controlflow.parent().dir())));
+                
+                Resources.copy(controlflow.file(), targetDir);
+            }
+        } else if (this.configFile != null && this.configFile.getResource() != null) {
+            Resources.copy(this.configFile.getFile(), Files.asResource(resourceLoader.getBaseDirectory()));
+        } else if (this.configFile != null && this.configFile.getProperties() != null) {
+            File controlFlowConfigurationFile = Resources.file(resourceLoader.get(PROPERTYFILENAME), true);
+            OutputStream out = Files.out(controlFlowConfigurationFile);
+            try {
+                this.configFile.getProperties().store(out, "");
+            } finally {
+                out.flush();
+                out.close();
+            }
+        }
+    }
+
+    @Override
+    public void loadConfiguration(GeoServerResourceLoader resourceLoader) throws IOException {
+        synchronized (this) {
+            Resource controlflow = resourceLoader.get(PROPERTYFILENAME);
+            if (Resources.exists(controlflow)) {
+                configFile = new PropertyFileWatcher(controlflow);
+                configFile.setKnownLastModified(System.currentTimeMillis());
```

Comment:
Done this way, isStale() is not going to return true after a configuration restore, or not? I may be wrong, but given it's already look at the files, it's likely that leaving this method empty will already do the trick... although looking at the FileWatcher code, it might fail if the file was there before, and gets removed later. But I don't see a reason to re-create the watcher otherwise.